src/TagManager/Cart.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace StefanDoorn\SyliusGtmEnhancedEcommercePlugin\TagManager;
+
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Currency\Context\CurrencyContextInterface;
+use Xynnn\GoogleTagManagerBundle\Service\GoogleTagManagerInterface;
+
+final class Cart implements CartInterface
+{
+    use CreateProductTrait;
+
+    private GoogleTagManagerInterface $googleTagManager;
+
+    private ChannelContextInterface $channelContext;
+
+    private CurrencyContextInterface $currencyContext;
+
+    private string $productIdentifier;
+
+    public function __construct(
+        GoogleTagManagerInterface $googleTagManager,
+        ChannelContextInterface $channelContext,
+        CurrencyContextInterface $currencyContext,
+        string $productIdentifier
+    ) {
+        $this->googleTagManager = $googleTagManager;
+        $this->productIdentifier = $productIdentifier;
+        $this->channelContext = $channelContext;
+        $this->currencyContext = $currencyContext;
+    }
+
+    public function getOrderItem(OrderItemInterface $orderItem): array
+    {
+        return $this->createProduct($orderItem, $this->productIdentifier);
+    }
+
+    public function add(array $productData): void
+    {
+        $products = [];
+        $products[] = $productData;
+
+        $this->googleTagManager->addPush([
+            'event' => 'addToCart',
+            'ecommerce' => [
+                'currencyCode' => $this->currencyContext->getCurrencyCode(),
+                'add' => [
+                    'products' => $products,
+                ],
+            ],
+        ]);
+    }
+
+    public function remove(array $productData): void
+    {
+        $products = [];
```

Comment:
I had to understand what's happening here, what about just removing these and then below where you use $products just do:

`'products' => [$productData]`?